astropy/table/tests/test_timeseries.py

```
@@ -0,0 +1,168 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import pytest
+from datetime import datetime
+
+from ...table import Table, QTable, Column
+from ..timeseries import TimeSeries, TimeSeriesColumn, TimeSeriesRow
+from ...time import Time, TimeDelta
+
+
+input_time = Time(['2016-03-22T12:30:31', '2015-01-21T12:30:32', '2016-03-22T12:30:40'])
+input_data = Table([[1, 2, 11], [3, 4, 1], [1, 1, 1]], names=['a', 'b', 'c'])
+
+
+@pytest.mark.xfail
+def test_empty_initialization():
+    ts = TimeSeries()
+    assert ts.time is None
+
+    assert isinstance(ts, TimeSeries)
+    assert isinstance(ts, QTable)
+
+
+@pytest.mark.xfail
+def test_initialization():
+    ts = TimeSeries(time=input_time, data=[[10, 2, 3], [4, 5, 6]], names=['a', 'b'])
+
+    assert isinstance(ts, TimeSeries)
+    assert isinstance(ts, QTable)
+
+    assert all(ts.time == input_time)
+
+    ts = TimeSeries(time=input_time, data=input_data)
+
+
+@pytest.mark.xfail
+def test_initialization_with_deltatime():
+    date = datetime(2018, 7, 1, 10, 10, 10)
+
+    input_time = Time(date) + TimeDelta(360, format='sec') * range(3)
+    ts = TimeSeries(time=input_time, data=[[10, 2, 3], [4, 5, 6]], names=['a', 'b'])
+
+    # TODO: we probably want to allow to avoid constructing the time array above and
+    #  initialize with the start time and step size
+
+
+@pytest.mark.xfail
+def test_initialization_with_time_in_data():
+    data = input_data.copy()
+    data['time'] = input_time
+
+    ts = TimeSeries(data=data)
+
+    # This should probably raise an error rather than deal with it
+    ts = TimeSeries(data=data, time=input_time)
```

Agree, this should raise an error - can you add a ``pytest.raises`` context?